### PR TITLE
feat(balances): extended the balances API 

### PR DESCRIPTION
**Pull request by DalderupMaurice**
Extended the balances API with an optional parameter to query other people their balance
Solves issue #82

### CODE REVIEW

File: src/actions/dapps/makeBalancesAction.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { createActions } from 'spunky';
+
+import getBalances from '../../util/getBalances';
+
+export const ID = 'balances_api';
+
+export default createActions(ID, ({ net, address, tokens = [], addedAddress } = {}) =>
```
Comment (mhuggins) [inline src/actions/dapps/makeBalancesAction.js]: The structure of this doesn't match the other action-makers under `actions/dapps`.  This should be wrapped in a function that takes a `sessionId` and `requestId`, which then builds a unique identifier for the redux store. i.e.:

```js
export const ID = 'balances';

export default function makeBalancesActions(sessionId, requestId) {
  const id = generateDAppActionId(sessionId, `${ID}-${requestId}`);

  return createActions(id, ({ net, address, tokens = [] }) => () => {
    return getBalances({ net, address, tokens })
  });
}
```

File: src/components/Browser/RequestsProcessor/GetBalance/index.js [hunk 2]
```diff
@@ -15,10 +15,18 @@ export default function makeGetBalance() {
   return compose(
```
Comment (mhuggins) [inline src/components/Browser/RequestsProcessor/GetBalance/index.js]: You'll need to update `makeGetBalances` to accept `balancesActions` instead, and then within `mappings.js` include `makeBalancesActions` there.

**Comment by DalderupMaurice** [comment]
Added PR comments although using the 2nd parameters (giving an address to fetch it's balance) does return 0. @mhuggins was going to have a look at the issue

File: src/actions/dapps/makeBalancesAction.js [hunk 3]
```diff
@@ -9,6 +10,10 @@ export default function makeBalancesActions(sessionId, requestId) {
   const id = generateDAppActionId(sessionId, `${ID}-${requestId}`);
 
   return createActions(id, ({ net, address, tokens = [] } = {}) => async () => {
+    if (!wallet.isAddress(address)) {
+      throw new Error(`Invalid address: "${address}".`);
+    }
```
Comment (DalderupMaurice) [inline src/actions/dapps/makeBalancesAction.js]: This is already done in getBalances 😄

Comment (mhuggins) [reply src/actions/dapps/makeBalancesAction.js]: Oops!  I'll drop remove this commit.

**Pull request merged**